Content.Server/Atmos/EntitySystems/BarotraumaSystem.cs

```diff
@@ -23,124 +24,122 @@ public sealed class BarotraumaSystem : EntitySystem
+            // any innate pressure resistance ?
+            if (TryComp<PressureProtectionComponent>(uid, out var innatePressureProtection))
+            {
+                barotrauma.HighPressureModifier += innatePressureProtection.HighPressureModifier;
+                barotrauma.HighPressureMultiplier *= innatePressureProtection.HighPressureMultiplier;
+                barotrauma.LowPressureModifier += innatePressureProtection.LowPressureModifier;
+                barotrauma.LowPressureMultiplier *= innatePressureProtection.LowPressureMultiplier;
+            }
         }
 
-        public float GetFeltHighPressure(EntityUid uid, BarotraumaComponent baro, float environmentPressure)
+        /// <summary>
+        /// Returns adjusted pressure after having applied resistances from equipment and innate (if any), to check against a low pressure hazard threshold
+        /// </summary>
+        public float GetFeltLowPressure(EntityUid uid, BarotraumaComponent barotrauma, float environmentPressure)
         {
-            var modifier = float.MinValue;
-            var multiplier = float.MinValue;
-
-            TryComp(uid, out InventoryComponent? inv);
-            TryComp(uid, out ContainerManagerComponent? contMan);
-
-            // TODO: cache this & update when equipment changes?
-            // Not as import and as low-pressure, but probably still useful.
-
-            if (baro.ProtectionSlots.Count == 0)
+            if (TryComp<PressureImmunityComponent>(uid, out var pressureImmunity))
             {
-                modifier = 0;
-                multiplier = 1;
+                return Atmospherics.OneAtmosphere;
             }
 
-            // First, check if for protective equipment
-            foreach (var slot in baro.ProtectionSlots)
-            {
-                if (!_inventorySystem.TryGetSlotEntity(uid, slot, out var equipment, inv, contMan)
-                    || !TryComp(equipment, out PressureProtectionComponent? protection))
-                {
-                    // Missing protection, skin is exposed.
-                    modifier = 0;
-                    multiplier = 1;
-                    break;
-                }
+            return (environmentPressure + barotrauma.LowPressureModifier) * (barotrauma.LowPressureMultiplier);
+        }
 
-                modifier = Math.Max(protection.HighPressureModifier, modifier);
-                multiplier = Math.Max(protection.HighPressureMultiplier, multiplier);
+        /// <summary>
+        /// Returns adjusted pressure after having applied resistances from equipment and innate (if any), to check against a high pressure hazard threshold
+        /// </summary>
+        public float GetFeltHighPressure(EntityUid uid, BarotraumaComponent barotrauma, float environmentPressure)
+        {
+            if (TryComp<PressureImmunityComponent>(uid, out var pressureImmunity))
+            {
+                return Atmospherics.OneAtmosphere;
             }
```

Comment:
Pressure immunity should also just be moved into the cached value. Just set the multiplier to 0 & modifier to one atmosphere

Reply:
Actually looking at the history that's more or less what you had before sloths comment. I'd say go back to that, but I agree with sloth that the field needs a comment & maybe rename it to "HasImmunity" rather than "Immuned"